src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -160,6 +160,11 @@ private void createLogEntry(LogEntry entry) throws LogServiceException {
                 // If the feedback session log type retrieved from the logs is invalid, discard it
                 continue;
             }
+
+            if (email != null && !email.equals(entryEmail)) {
+                // Retrieved log entry does not belong to the email to filter for
+                continue;
+            }
```

Comment:
I think this could be unnecessary? In the case that `email` is not null, then it should be the same as `entryEmail` (because the email was filtered for previously when searching for logs)